lib/web_components_rails/html_import_processor.rb

```diff
@@ -24,22 +67,22 @@ def self.doc_to_html(doc)
     # Nokogiri/Nokogumbo are hard-coded to URI-escape certain attributes (src, href, action, and a[name]),
     # so we have to put in placeholders, and fix the values in the HTML string output afterwards
     # This doesn't work so well with framework-specific syntax (eg. <foo src="{{bar}}">)
-    placeholder_mapping = {}
-    %w(src href action).each do |name|
-      doc.css("[#{name}]").each do |node|
-        # The placeholders are just random strings
-        placeholder = SecureRandom.hex(40)
-        attr = node.attributes[name]
-        placeholder_mapping[placeholder] = attr.value
-        attr.value = placeholder
-      end
-    end
-    new_html = doc.to_html
-    placeholder_mapping.each do |placeholder, value|
-      new_html.sub!(placeholder, value)
+    doc_html = doc.to_html(encoding: 'UTF-8')
+
+    uri_regex = '(' + URI_ATTRIBUTES.join('|') + ')'
+    square_brackets_regex = Regexp.new("#{uri_regex}=\"%5B%5B(.+?)%5D%5D\"", 'i')
+    curly_braces_regex = Regexp.new("#{uri_regex}=\"%7B%7B(.+?)%7D%7D\"", 'i')
```

Comment:
I think this makes sense, and may be cleaner than the old way, but you put a comment explaining what this section is doing?